### PR TITLE
Add updateUiSchema form system helper

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
This helper allows you to update the uiSchema as the form data changes. It works just like the ui:options.updateSchema callback. I'm using this on the revised profile address form so that the country dropdown can be disabled when the user has selected that they live on a military base outside of the US. (Disabling is done via `ui:disabled` on the uiSchema)

GIF of an example using the new updateUiSchema handler to disable a select field:
disable

## Testing done
Local + new unit tests

## Acceptance criteria
- [ ] updating the uiSchema on the fly works
- [ ] it doesn't break anything else

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by billfienberg** [comment]
Thank you for including tests with this. The code makes sense, but I feel like I should defer to somebody that has more experience with the form system, like @cvalarida.

**Review by cvalarida — COMMENTED** [review]
:thinking: How will this get used in the forms system?

I looks like it's used in a custom thingy for the [vet360 PR](https://github.com/department-of-veterans-affairs/vets-website/pull/11601/files#diff-b9df097e73755d3f01938796c45c2b8dR46). Is the plan to make this available for all forms, or just this one?

**Comment by erikphansen** [comment]
@cvalarida It doesn't _have to_ get used by the form system if that's not desirable. I felt like making a significant change to all the benefits forms was a little bit of a reach for this right now. We could add a call to this new function in `updateSchemaAndData()` but we expect that function to return an object with two keys and _not_ return a new uiSchema object. So it would be a larger discussion to determine the best place/way to make this part of the normal data update flow.

File: src/platform/forms-system/src/js/state/helpers.js [hunk 1]
```diff
@@ -349,6 +349,53 @@ export function updateSchemaFromUiSchema(
...
+ * A helper that returns a new uiSchema based on the input uiSchema and formData
+ * @param {Object} uiSchema - The uiSchema to update
+ * @param {Object} formData - The form data to based uiSchema updates on
+ * @returns {Object} The new uiSchema object
+ */
+export function updateUiSchema(uiSchema, formData) {
+  let currentUiSchema = uiSchema;
+
+  if (typeof currentUiSchema === 'object') {
+    const newUiSchema = Object.keys(currentUiSchema).reduce((current, next) => {
```
[the remark this answers is not in the document]
Comment (erikphansen) [reply src/platform/forms-system/src/js/state/helpers.js]: sure. This was just copied from `updateSchemaFromUiSchema` :)

**Comment by cvalarida** [comment]
Ohh, gotcha. Yeah, I think having it be a part of the normal flow would be great! I'm not really sure what the limitations are there, though. (It's been a while since I dug into that part of the code.)

If you don't want to implement that here, could you mention that it isn't available in the `formConfig` yet in the comment for the function and write a ticket for us to pick up later?

Alternatively, you could pull this helper out into the `vet360` directory (or wherever makes sense) so it's not in `forms-system` confuse the next passerby.

**Comment by erikphansen** [comment]
@cvalarida I'll update the docs and write a ticket pointing to this function to discuss how to best work it into a typical benefit application app when this is merged in the codebase.

File: src/platform/forms-system/src/js/state/helpers.js [hunk 2]
```diff
@@ -349,6 +349,65 @@ export function updateSchemaFromUiSchema(
...
+/**
+ * A helper that returns a new uiSchema based on the input uiSchema and
+ * formData. Only updates and returns a uiSchema object for the field that has
+ * an `updateUiSchema` callback defined in its `ui:options`.
+ *
+ * Note that this helper is _not_ called as part of the normal data-update flow
+ * in a standard form config-powered benefit application. Perhaps it could be
+ * called as part of updateSchemaAndData() (also in this file) but we'd have to
+ * investigate any possible side effects of updating that function to add a
+ * uiSchema prop in the object it returns.
```
Comment (cvalarida) [inline src/platform/forms-system/src/js/state/helpers.js]: :100: Thanks for adding this.

**Review by cvalarida — APPROVED** [review]

**Comment by erikphansen** [comment]
Ticket to support this in all standard form system apps: https://github.com/department-of-veterans-affairs/va.gov-team/issues/5577

**Pull request merged**